packages/@aws-cdk/aws-apigateway/lib/authorizers/identity-source.ts

```diff
@@ -0,0 +1,49 @@
+import { Token } from '@aws-cdk/core';
+
+/**
+ * Represents an identity source.
+ *
+ * The source can be specified either as a literal value (e.g: `Auth`) which
+ * cannot be blank, or as an unresolved string token.
+ */
+export class IdentitySource {
+  /**
+   * @param headerName the name of the header the `IdentitySource` will represent.
+   * @returns a new `IdentitySource` representing `headerName` from the request.
+   */
+  public static header(headerName: string): IdentitySource {
```

Comment:
Curious: shouldn't the return type here be `string`? How does typescript accept `IdentitySource` as a `string` type at the point of assignment?

Reply:
I think TS is inferring that `IdentitySource` is a compatible string-type, but definitely better to revert to just string.